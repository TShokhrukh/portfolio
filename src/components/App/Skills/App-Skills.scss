.App-Skills {
  padding: $spacer * 20 0;
}

@include media-breakpoint-up(xl) {
  .App-SkillsContainer {
    max-width: 920px;
  }
}