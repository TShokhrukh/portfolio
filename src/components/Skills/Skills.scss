.Skills {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.Skills-Title {
  margin-top: $spacer * 3;
}