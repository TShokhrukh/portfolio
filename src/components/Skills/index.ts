export * from './build'
export * from './__types__'
export { Skills as SkillsBase } from './Skills'
