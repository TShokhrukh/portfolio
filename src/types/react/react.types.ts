import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
