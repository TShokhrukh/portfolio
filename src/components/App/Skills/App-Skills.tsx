import React from 'react'
import { Container } from '~/components/Container/build'
import { Skills } from '~/components/Skills/build'
import { cnApp } from '../App.const'
import { useAppContext } from '../libs/context'
import './App-Skills.scss'

export const AppSkills: React.FC = () => {
  const { skills } = useAppContext()
  return (
    <section id='skills' className={cnApp('Skills')}>
      <Container className={cnApp('SkillsContainer')}>
        <Skills skills={skills} />
      </Container>
    </section>
  )
}
