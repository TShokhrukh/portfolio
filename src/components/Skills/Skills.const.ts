import { createContext } from 'react'
import { cn } from '@bem-react/classname'
import type { ISkillsContext } from './__types__'

export const cnSkills = cn('Skills')

export const Context = createContext<ISkillsContext>(null!)
