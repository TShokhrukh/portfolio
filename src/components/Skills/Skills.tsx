import React, { FC } from 'react'
import { Text } from '@eo-locale/react'
import { BadgeSection } from '~/components/BadgeSection/build'
import { SkillsContent } from './Content'
import { cnSkills, Context } from './Skills.const'
import type { ISkillsProps } from './__types__'
import './Skills.scss'

export const Skills: FC<ISkillsProps> = ({
  skills,
  className,
  ...props
}) => {
  return (
    <Context.Provider value={{ skills }}>
      <BadgeSection {...props} badge={<Text id='skills.badge' />} title={<Text id='skills.title' />}
        className={cnSkills({}, [className])}>
        <SkillsContent />
      </BadgeSection>
    </Context.Provider>
  )
}
