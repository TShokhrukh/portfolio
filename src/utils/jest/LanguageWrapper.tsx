import React, { ReactNode } from 'react'
import { TranslationsProvider } from '@eo-locale/react'

import type { Locale } from '@eo-locale/core'

interface IWrapperProps {
  children: ReactNode,
  language?: string,
  locales?: Locale[]
}

const defLocales: Locale[] = [
  {
    language: 'en',
    messages: {}
  }
]

export const LanguageWrapper: React.FC<IWrapperProps> = ({
  children,
  language = 'en',
  locales = defLocales
}) => {
  return (
    <TranslationsProvider language={language} locales={locales} onError={() => null}>
      {children}
    </TranslationsProvider>
  )
}
