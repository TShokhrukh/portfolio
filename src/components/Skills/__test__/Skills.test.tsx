/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { LanguageWrapper } from '~/utils/jest/LanguageWrapper'
import { Skills } from '../Skills'
import type { ISkill } from '../__types__'

describe('Skills', () => {
  test('must return the component template', () => {
    const skills: ISkill[] = [
      {
        id: '1',
        img: 'http://localhost:9000/img/1.png'
      },
      {
        id: '2',
        img: 'http://localhost:9000/img/2.png'
      }
    ]

    render(
      <LanguageWrapper>
        <Skills skills={skills} className='test_class' data-testid='skills' />
      </LanguageWrapper>
    )

    expect(screen.getByTestId('skills')).toMatchSnapshot()
  })
})
