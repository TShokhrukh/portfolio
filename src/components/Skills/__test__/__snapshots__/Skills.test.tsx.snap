// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills must return the component template 1`] = `
<div
  class="Skills test_class"
  data-testid="skills"
>
  <div>
    <span
      class="Badge Badge_color_purple"
    >
      skills.badge
    </span>
  </div>
  <h2
    class="Typography Typography_color_white Typography_variant_h2 Skills-Title"
  >
    skills.title
  </h2>
  <div
    class="List Skills-Content"
  >
    <img
      alt="..."
      class="Img Skills-Item"
    />
    <img
      alt="..."
      class="Img Skills-Item"
    />
  </div>
</div>
`;
